smaug/operators/padding_op.h

```diff
@@ -0,0 +1,115 @@
+#ifndef _OPERATORS_PADDING_OP_H_
+#define _OPERATORS_PADDING_OP_H_
+
+#include "smaug/core/backend.h"
+#include "smaug/core/operator.h"
+#include "smaug/core/tensor.h"
+// #include "smaug/core/tensor_utils.h"
+#include "smaug/core/workspace.h"
+
+namespace smaug {
+
+/** \ingroup Operators
+ * \brief Pad a given tensor in different dimension.
+ *
+ * This has a software-based implementation.
+ *
+ * @tparam Backend The Backend that sets Alignment.
+ */
+template <typename Backend>
+class PaddingOp : public Operator {
+   public:
+    PaddingOp(const std::string& name,
+              Workspace* workspace)
+            : Operator(name, OpType::Repeat, workspace){
+        inputs.resize(kNumInputs, nullptr);
+        outputs.resize(kNumOutputs, nullptr);
+    }
+
+    PaddingOp(const std::string& name,
+              Workspace* workspace,
+              int val)
+            : Operator(name, OpType::Repeat, workspace), padder(val){
+        inputs.resize(kNumInputs, nullptr);
+        outputs.resize(kNumOutputs, nullptr);
+    }
+
+    /** Set the number of padders of the Tensor along each dimension. */
+    void setPadder(const int& val) {
```

Comment:
I think it would be relatively straightforward to make this support tensors with any number of dimensions and arbitrary padding size along all of them, so we should do that instead. The `copyTensorRegion` API should work regardless.

Reply:
Good idea. I'll rename the operation.

Reply:
The logic modification seems straightforward. But I can't set the `paddingSize` properly in protobuf. I use `std::vector<int> paddingSize` in CPP code. And in protobuf it uses `repeated int32 padding_size = 1;`  But it seems does not work.